### DISCUSSION TITLE
Groups anchors

### DISCUSSION THREAD

**Discussion by Robotechnic** [cetz-package/cetz#686] [Q&A] [answered]
Hello,
I have a problem with the groups anchors and I don't know if it's a normal behaviour.
Here is a minimal example:
```typ
#import "@preview/cetz:0.2.2"

#cetz.canvas({
	import cetz.draw: *
	group(name: "mol2", anchor: "south",{
		line((0,0),(2,2))
	})
	circle("mol2.east", radius: .1em, fill: red, stroke: red)
	circle("mol2.west", radius: .1em, fill: red, stroke: red)
	circle("mol2.north", radius: .1em, fill: red, stroke: red)
	circle("mol2.south", radius: .1em, fill: red, stroke: red)
	circle("mol2.center", radius: .1em, fill: red, stroke: red)
	circle((0,0), fill: blue, stroke: blue, radius: .1em)
})
```
This gives:
![image](https://github.com/user-attachments/assets/5da2f616-463c-43ab-9d7a-93f8f750cfe0)

I may have misunderstood how the anchor argument works but, for me, the blue dot should be on the bottommost red dot.

Here is the same thing but with "north"
![image](https://github.com/user-attachments/assets/685dadef-66f2-4534-a37c-cefdc761953e)

"east"
![image](https://github.com/user-attachments/assets/34b6b5c3-e02f-40e5-be95-e6f02eba711a)

"west"
![image](https://github.com/user-attachments/assets/5310e040-6ae5-45e6-b6b9-f5bf2492e6e6)

We can see in the images above that the blue dot would be in the right place if the start of the line was the group's center.

**Comment by fenjalien** [comment]
Normally elements are drawn positoined on their default anchor, in this case for a group its the center anchor. The `anchor` parameter causes the element to be translated such that the specified anchor is in the position of where the default would be. In your first picture the group is being translated up such that the south anchor is where the center of the group element.

**Reply by Robotechnic** [reply, marked as answer]
Ok, thanks. Adding a default anchor to (0,0) does what I want.
